lzy/tunnel-agent/src/main/java/ai/lzy/tunnel/service/LzyTunnelAgentService.java

```diff
@@ -1,54 +1,75 @@
 package ai.lzy.tunnel.service;
 
+import ai.lzy.util.grpc.ProtoPrinter;
 import ai.lzy.v1.tunnel.LzyTunnelAgentGrpc;
 import ai.lzy.v1.tunnel.TA;
+import io.grpc.Status;
 import io.grpc.stub.StreamObserver;
+import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
 
 @Singleton
 public class LzyTunnelAgentService extends LzyTunnelAgentGrpc.LzyTunnelAgentImplBase {
 
     private static final Logger LOG = LogManager.getLogger(LzyTunnelAgentService.class);
 
-    public static final String TUNNEL_SCRIPT_PATH = "/app/resources/scripts/tunnel.sh";
+    private final TunnelManager tunnelManager;
+
+    @Inject
+    public LzyTunnelAgentService(TunnelManager tunnelManager) {
+        this.tunnelManager = tunnelManager;
+    }
 
     @Override
     public void createTunnel(TA.CreateTunnelRequest request, StreamObserver<TA.CreateTunnelResponse> responseObserver) {
-        LOG.info(
-            "Create Tunnel, remote v6 {}, worker pod v4 {}, k8s pod cidr {}",
-            request.getRemoteV6Address(),
-            request.getWorkerPodV4Address(),
-            request.getK8SV4PodCidr()
-        );
-        try {
-            Process process = Runtime.getRuntime().exec(
-                new String[]{
-                    TUNNEL_SCRIPT_PATH,
-                    request.getRemoteV6Address(),
-                    request.getWorkerPodV4Address(),
-                    request.getK8SV4PodCidr(),
-                    "2>&1"
-                }
-            );
-            BufferedReader reader = new BufferedReader(
-                new InputStreamReader(process.getInputStream())
-            );
-            String s;
-            while ((s = reader.readLine()) != null) {
-                // TODO: think about better script output logging
-                LOG.info(s);
-            }
-        } catch (IOException e) {
-            LOG.error("Error reading output of script", e);
-            responseObserver.onError(e);
+        LOG.info("Create Tunnel: {}", ProtoPrinter.printer().shortDebugString(request));
```

Comment:
if request contains any secrets then use `.safePrinter()` instead of `.printer()`

Reply:
There's no secrets in this area, but I've replaced printer just in case :)